var/spack/repos/builtin/packages/genemark-et/package.py

```diff
@@ -27,12 +27,17 @@ class GenemarkEt(Package):
     def url_for_version(self, version):
         return "file://{0}/gm_et_linux_64.tar.gz".format(os.getcwd())
 
+    def patch(self):
+        with working_dir(join_path(self.stage.source_path, "gmes_petap")):
+            perlscripts = glob.glob('*.pl')
+            filter_file('#!/usr/bin/perl', '#!/usr/bin/env perl', *perlscripts)
+
     def install(self, spec, prefix):
         mkdirp(prefix.bin)
         mkdirp(prefix.bin.heu_dir)
         with working_dir('gmes_petap'):
             install_tree('lib', prefix.lib)
-            files = glob.iglob('*')
+            files = glob.iglob('*[!~]')
```

Comment:
could you add a comment explaining the intent? It took me a while to understand you are installing all files that don't end in `~`. This is primarily because the documentation on `glob` itself isn't very clear on the syntax (which is not the same as regex, which uses `^` for negation in character classes).